launcher/src/main/java/org/apache/spark/launcher/SparkSubmitOptionParser.java

```diff
@@ -76,6 +76,12 @@
   protected final String PRINCIPAL = "--principal";
   protected final String QUEUE = "--queue";
 
+  // Kubernetes-only options.
+  protected final String KUBERNETES_MASTER = "--kubernetes-master";
+  protected final String KUBERNETES_NAMESPACE = "--kubernetes-namespace";
+  protected final String KUBERNETES_UPLOAD_JARS = "--upload-jars";
```

Comment:
I wonder if, in the case where the user expects jars on the docker image, they can also be assumed responsible for making sure their final CMD invocation knows where they are.  IIUC, it would imply they put extra jars there, via defining their own images